webcrypto/crypto.go

```diff
@@ -1,16 +1,94 @@
 package webcrypto
 
-import "go.k6.io/k6/js/modules"
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
 
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modules"
+)
+
+// Crypto represents the Crypto interface of the Web Crypto API.
 type Crypto struct {
 	vu modules.VU
 
 	Subtle    *SubtleCrypto      `json:"subtle"`
 	CryptoKey *CryptoKey[[]byte] `json:"CryptoKey"`
 }
 
-func (c *Crypto) GetRandomValues() {
-	// TODO
+// QuotaExceededError is the error thrown if the byteLength of a typedArray
+// exceeds 65,536.
+var QuotaExceededError = errors.New("QuotaExceededError")
+
+// TypeMismatchError is the error thrown if the typedArray given to
+// getRandomValues is not of a supported type.
+var TypeMismatchError = errors.New("TypeMismatchError")
+
+// GetRandomValues lets you get cryptographically strong random values.
+//
+// Do not generate keys using the getRandomValues method. Use the generateKey method instead.
+//
+// The array given as the parameter is filled with random numbers (random in
+// its cryptographic sense, not in its statistical sense).
+//
+// To guarantee enough performance, this implementation is not using a truly
+// random number generator, but is using a pseudo-random number generator
+// seeded with a value with enough entropy. We are using the golang
+// crypto/rand package, which uses the operating system's random number
+// generator.
+func (c *Crypto) GetRandomValues(typedArray goja.Value) goja.Value {
+	// 1.
+	// If the provided typedArray is not of an integer type (Int8Array, Uint8Array,
+	// Int16Array, UInt16Array, Int32Array, UInt32Array,  Uint8ClampedArray), we
+	// throw a TypeMismatchError.
+	if !IsInstanceOf(c.vu.Runtime(), typedArray, Int8ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Uint8ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Int16ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Uint16ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Int32ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Uint32ArrayConstructor) &&
+		!IsInstanceOf(c.vu.Runtime(), typedArray, Uint8ClampedArrayConstructor) {
+		common.Throw(c.vu.Runtime(), TypeMismatchError)
+	}
+
+	// 2.
+	// Obtain the length of the typed array, and throw a QuotaExceededError if
+	// it's too big, as specified in the [spec's] 10.2.1.2 paragraph.
+	// [spec]: https://www.w3.org/TR/WebCryptoAPI/#Crypto-method-getRandomValues
+	obj := typedArray.ToObject(c.vu.Runtime())
+	objLength := obj.Get("length").ToNumber().Export().(int64)
+	if objLength > 65536 {
```

Comment:
Could you please move the "magic" number `65536` to a const with some explicit name?

Reply:
Fair point, wilco 👍🏻